lib/ui/shared/icon_button_with_label.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:paintroid/tool/tool.dart';
+import 'package:toast/toast.dart';
+
+class IconButtonWithLabel extends StatelessWidget {
```

Comment:
We are mixing too much UI with logic in here.
Could we make this widget 100% generic and hand over the function as a callback and remove the tooldata?

If you do not like that solution, renaming it to something more specific is fine for me too.
But now it seems like a very generic widget, but it changes the tool

The 100% perfect solution would be a very generic IconButtonWithLabel that gets icon, label & callback
And then a ToolButton which uses the IconButtonWithLabel and implements the tool switching

Reply:
Create a new ToolButton widget, moved toast to provider,